source/frontend/src/features/documents/list/DocumentResults/DocumentResultsColumns.tsx

```diff
@@ -112,46 +132,103 @@ const renderActions = (
-        {hasClaim(Claims.DOCUMENT_DELETE) && (
-          <Col>
+
+    const documentInError =
+      (original.mayanDocumentId === null || original.mayanDocumentId === -1) &&
+      original.queueStatusTypeCode &&
+      (original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.PIMS_ERROR ||
+        original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.MAYAN_ERROR);
+
+    const documentProcessing =
+      (original.mayanDocumentId === null || original.mayanDocumentId < 1) &&
+      original.queueStatusTypeCode &&
+      (original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.PENDING ||
+        original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.PROCESSING);
+
+    const canViewDocument =
+      (original.mayanDocumentId &&
+        original.mayanDocumentId > 0 &&
+        original.queueStatusTypeCode === null) ||
+      (original.queueStatusTypeCode &&
+        original.queueStatusTypeCode.id === ApiGen_CodeTypes_DocumentQueueStatusTypes.SUCCESS &&
+        original.mayanDocumentId &&
+        original.mayanDocumentId > 0);
+
+    if (documentProcessing) {
+      return (
+        <DocumentsActionsDiv>
+          <FaClock
+            id={`document-processing-${index}`}
+            size={21}
+            title="Upload in progress..."
+            className="warning"
+          />
+        </DocumentsActionsDiv>
+      );
+    }
+
+    if (documentInError) {
+      return (
+        <DocumentsActionsDiv>
+          <FaTimesCircle
+            id={`document-error-${index}`}
+            size={21}
+            title={original.queueStatusTypeCode.description}
+          />
+
+          {hasClaim(Claims.DOCUMENT_DELETE) && (
             <StyledRemoveLinkButton
-              title="document delete"
               data-testid="document-delete-button"
-              icon={<FaTrash size={24} id={`document-delete-${index}`} title="document delete" />}
+              icon={<FaTrash id={`document-delete-${index}`} size={21} title="document delete" />}
               onClick={() => original?.id && onDelete(DocumentRow.toApi(original))}
             ></StyledRemoveLinkButton>
-          </Col>
+          )}
+        </DocumentsActionsDiv>
+      );
+    }
+
+    return (
+      <DocumentsActionsDiv>
+        {hasClaim(Claims.DOCUMENT_VIEW) && canViewDocument && (
+          <Button
+            data-testid="document-view-button"
+            icon={<FaEye size={21} id={`document-view-${index}`} title="document view details" />}
+            onClick={() => original?.id && onViewDetails(DocumentRow.toApi(original))}
+          ></Button>
+        )}
+
+        {hasClaim(Claims.DOCUMENT_DELETE) && (
+          <StyledRemoveLinkButton
+            data-testid="document-delete-button"
+            icon={<FaTrash size={21} id={`document-delete-${index}`} title="document delete" />}
+            onClick={() => original?.id && onDelete(DocumentRow.toApi(original))}
```

Comment:
I'd prefer to not display the button if there is no id, instead of displaying a button that does nothing when clicked.

Reply:
A little lost here, the only way this displays is when the document has been processed as either success or failure. if it fails it deletes everything so they can retry.

Reply:
right now when the button is checked you ensure that original?.id is set before you fire the call to onDelete. I'd rather you either remove that check, or hide the entire button if original?.id is not set.